src/hooks/useLocation.jsx

```diff
@@ -0,0 +1,20 @@
+import {useEffect, useState} from "react";
+
+export function useLocation() {
+    const [location, setLocation] = useState({});
+
+    useEffect(() => {
+        getLocation();
+    }, []);
```

Comment:
if` navigator.geolocation.getCurrentPosition` not async its redundant as well

Reply:
navigator.geolocation.getCurrentPosition async